repose-aggregator/components/filters/body-extractor-to-header/src/main/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,152 @@
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.bodyextractortoheader.config.BodyExtractorToHeaderConfig
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+@Named
+class BodyExtractorToHeaderFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[BodyExtractorToHeaderConfig] with LazyLogging {
+
+  import BodyExtractorToHeaderFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var extractions: Iterable[Extraction] = _
+  val jsonPathConfiguration = JsonConfiguration.defaultConfiguration()
+  jsonPathConfiguration.addOptions(JsonOption.DEFAULT_PATH_LEAF_TO_NULL)
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("Body Extractor to Header filter initializing...")
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info("Initializing Body Extractor to Header filter using config {}", configurationFile)
+    val xsdUrl: URL = getClass.getResource(SCHEMA_FILE_NAME)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFile, xsdUrl, this, classOf[BodyExtractorToHeaderConfig])
+
+    logger.trace("Body Extractor to Header filter initialized.")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+
+    def addHeader(name: String, value: String, quality: Option[java.lang.Double], overwrite: Boolean): Unit = {
+      if (overwrite) {
+        mutableHttpRequest.removeHeader(name)
+      }
+      val hdrValue = quality match {
+        case Some(qual) => s"$value;q=$qual"
+        case None => value
+      }
+      mutableHttpRequest.addHeader(name, hdrValue)
+    }
+
+    val jsonDoc: Try[DocumentContext] = {
+      Option(mutableHttpRequest.getContentType) match {
+        case Some(contentType) =>
+          if (contentType.toLowerCase.contains("json")) {
+            val is = mutableHttpRequest.getInputStream
+            if (is.markSupported()) is.mark(0)
+            val jsonString = Source.fromInputStream(is).mkString
+            if (is.markSupported()) is.reset()
+            logger.trace(s"is.available = ${is.available()}")
+            Try(JsonPath.using(jsonPathConfiguration).parse(jsonString))
+          } else {
+            new Failure(new Exception)
+          }
+        case _ =>
+          logger.debug("Content-Type was not defined.")
+          new Failure(new Exception)
+      }
+    }
+
+    extractions.foreach { extraction =>
+      if (jsonDoc.isSuccess) {
+        val extracted = jsonDoc match {
+          case Success(doc) => Try(doc.read[Any](extraction.jsonPath))
```

Comment:
Why do the `jsonDoc.isSuccess` check if you're going to match here anyway? Just stick everything in a match. Of course, that means you'll have to handle the other case, but it can just do nothing as it does now.

Also, match will probably be cleared if you switch `jsonDoc` to an `Option[Try[DocumentContext]`.

```
jsonDoc match {
  case Some(Success(doc)) => 
    val extracted = Try(doc.read[Any](extraction.jsonPath))
    ...
  case _ => 
    // do nothing
}
```